R/mod_03_plot_data.R

```diff
@@ -135,137 +139,131 @@ mod_03_plot_data_server <- function(id) {
+          mapProxy |>
+            addRiverflySpeciesMarkers(
+              popup_data = riverfly_species_popups,
+              map_data = Riverfly_Species_Plot_Recent[[selectedTaxa]],
+              selectedTaxa,
+              screen_width()
+            ) |>
+            showGroup("Riverfly points")
+        } else if (selected_riverfly() == "Other species") {
+          # If the user chooses Other species, plot abundance data
+          # Filter data for the selected 'other species' from the radio buttons
+          selectedTaxa <- names(which(
+            other_spp_bw == selected_other_species()
+          ))
+          otherspeciesData_Recent_Map <- Riverfly_Other_Species_Plot_Recent[[
+            selectedTaxa
+          ]]
+          mapProxy |>
+            addOtherSpeciesMarkers(
+              otherspeciesData_Recent_Map,
+              selectedTaxa
+            ) |>
+            showGroup("Other spp points")
+        }
+      } else if (selected_metric() == "Invasive Species") {
         # If the user chooses Invasive Species, plot presence/absence data
-        BRCInvSpcs_Plot_Recent <- make_recent_inv_spp(
-          BRCInvSpcs,
-          BRC_locs,
-          plot_palette
-        )
-        addInvasiveSpeciesMarkers(
-          mapProxy,
-          BRCInvSpcs_Plot_Recent,
-          input$invasiveType,
-          rev(brewer.pal(n = 4, name = "Blues"))
-        )
-      } else if (input$metric == "Water Chemistry") {
+        mapProxy |>
+          addInvasiveSpeciesMarkers(
+            BRCInvSpcs_Plot_Recent,
+            selected_invasive_type(),
+            rev(brewer.pal(n = 4, name = "Blues"))
+          ) |>
+          showGroup("Invasive points")
+      } else if (selected_metric() == "Water Chemistry") {
         # If the user chooses Water Chemistry, plot water quality data
 
-        # If the user chooses Water Chemistry, plot water quality data
-        WQ_plot_data <- make_water_quality_plot_data(
-          BRC_wq,
-          Unique_BRC_Sampling_Locs
-        )
-        selectedReading <- input$readingType
-        wq_Recent_Map <- WQ_plot_data$recent |>
-          filter(reading_type == selectedReading)
+        wq_Recent_Map <- WQ_plot_data$recent[[selected_reading_type()]]
 
-        wq_data <- WQ_plot_data$all_obs |>
-          filter(reading_type == selectedReading)
+        wq_data <- WQ_plot_data$all_obs[grepl(
+          selected_reading_type(),
+          names(WQ_plot_data$all_obs)
+        )]
 
-        addWaterQualityMarkers(
-          wq_data = wq_data,
-          wq_data_recent = wq_Recent_Map,
-          mapProxy = mapProxy,
-          input = input
-        )
+        mapProxy |>
+          addWaterQualityMarkers(
+            wq_data = wq_data,
+            wq_data_recent = wq_Recent_Map,
+            metric = selected_reading_type(),
+            screen_width = screen_width()
+          )
+
+        mapProxy |>
+          showGroup("Water Quality points")
       }
+      # mapProxy |>
```

Comment:
Why is section commented-out?

Reply:
This was used for debugging but I don't want it in the final app - I'll remove it once I'm done with the performance attempts.